test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,334 @@
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/search/algorithm/all.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+struct options
+{
+    size_t const sequence_length;
+    bool const has_repeats;
+    size_t const number_of_reads;
+    size_t const read_length;
+    double const prob_insertion;
+    double const prob_deletion;
+    uint8_t const simulated_errors;
+    uint8_t const searched_errors;
+    uint8_t const strata;
+    double const standard_deviation = 0;
+    uint32_t repeats = 20;
+};
+
+template <Alphabet alphabet_t>
+void mutate_insertion(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<uint8_t> dis_alpha{0, alphabet_size<alphabet_t> - 1};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    seq.insert(std::ranges::begin(seq) + random_pos(gen), alphabet_t{}.assign_rank(dis_alpha(gen)));
+}
+
+template <Alphabet alphabet_t>
+void mutate_deletion(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    seq.erase(std::ranges::begin(seq) + random_pos(gen));
+}
+
+template <Alphabet alphabet_t>
+void mutate_substitution(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<uint8_t> dis_alpha_short{0, alphabet_size<alphabet_t> - 2};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    alphabet_t & cbase = seq[random_pos(gen)];
+    uint8_t crank = to_rank(cbase);
+    uint8_t rrank = dis_alpha_short(gen);
+    if (rrank >= crank)
+        ++rrank;
+    cbase.assign_rank(rrank);
+}
+
+template <Alphabet alphabet_t>
+auto generate_reads(std::vector<alphabet_t> const & ref,
+                    size_t const number_of_reads,
+                    size_t const read_length,
+                    uint8_t const simulated_errors_,
+                    double const prob_insertion,
+                    double const prob_deletion,
+                    double const standard_deviation = 0,
+                    size_t const seed = 0)
+{
+    std::vector<std::vector<alphabet_t>> reads;
+    std::mt19937 gen{seed};
+    std::uniform_int_distribution<size_t> seeds{0};
+    std::normal_distribution<> dis_error_count{static_cast<double> (simulated_errors_), standard_deviation};
```

Comment:
You don't need to `static_cast`, `std::normal_distribution` will always return floating point types.
E.g. https://coliru.stacked-crooked.com/view?id=bc43dd8bec4be9c8